### DISCUSSION TITLE
mconcat on semigroups

### DISCUSSION THREAD

**Discussion by viglioni** [gcanti/fp-ts#1824] [Ideas]
Ciao! How are you?

I was wondering if adding `mconcat` functions on SemiGroups using `ReadonlyArrays`  would be something that makes sense in the library

```typescript
import { reduce } from 'fp-ts/lib/ReadonlyArray'

interface Monoid<A> extends Semigroup<A> {
  readonly empty: A,
  readonly mconcat: ReadonlyArray<A> => A
}

// Example of monoid sum implementation
const mconcat = reduce(MonoidSum.empty, MonoidSum.concat)
```

(it's not a feature request exactly, I can try to create a PR with this new feat, I just want to know if this is desirable)

**Comment by gcanti** [comment]
There's [`concatAll`](https://gcanti.github.io/fp-ts/modules/Monoid.ts.html#concatall)

**Reply by viglioni** [reply]
grazie!
